s3/pkg/datastore/gcp/gcp.go

```diff
@@ -39,9 +39,30 @@ type GcsAdapter struct {
 	session *s3client.Client
 }
 
+func (ad *GcsAdapter) BucketDelete(ctx context.Context, input *pb.Bucket) error {
+	bucket := ad.session.NewBucket()
+	err := bucket.Remove(input.Name)
+	if err != nil {
+		log.Error("falied to delete bucket", err)
+	}
+	log.Info("bucket deletion successful")
```

Comment:
Please handle the cenario, incase bucket deletion is uncussessful

Reply:
updated

Reply:
updated